soa-iterator.cc

```diff
@@ -7,32 +7,44 @@ using std::endl;
 using std::vector;
 using std::for_each;
 
+struct element {
+    //TODO: add all the relevant constructors.
+    double& x() { return points_.get_x()[index_]; }
+    double& y() { return points_.get_y()[index_]; }
+    Points* points_ = nullptr;
+    //TODO: perhaps make this private and give iterator friendship relations
+    std::size_t index_ = 0; 
+}; //end struct element
+
 class Points {
+ typedef std::vector< double> Vector;
  public:
 
-  typedef vector<double>::size_type size_type;
+  typedef typename Vector::size_type size_type;
 
   class iterator {
```

Comment:
Ryan, do you think the iterator class should be defined outside of the container class?